openerp/tools/amount_to_text.py

```diff
@@ -156,11 +156,110 @@ def amount_to_text_nl(number, currency):
+             u'Quintilhões', u'Sextilhões', u'Septilhões', u'Octilhões',
+             u'Nonilhões', u'Decilhões', u'Undecilhões', u'Duodecilhões',
+             u'Tredecilhões', u'Quattuordecilhões', u'Quindecilhões',
+             u'Sexdecilhões', u'Septendecilliões', u'Octodecilhões',
+             u'Novendecilhões', u'Vigintilhões')
+
+
+def _convert_nn_br(val):
+    """ convert a value < 100 to Brazilian Portuguese
+    """
+    if val < 20:
+        return to_19_br[val]
+    for (dcap, dval) in ((k, 20 + (10 * v)) for (v, k) in enumerate(tens_br)):
+        if dval + 10 > val:
+            if val % 10:
+                return dcap + ' e ' + to_19_br[val % 10]
+            return dcap
+
+
+def _convert_nnn_br(val):
+    """ convert a value < 1000 to Brazilian Portuguese
+        special cased because it is the level that kicks
+        off the < 100 special case.  The rest are more general.
+        This also allows you to get strings in the form of
+        'forty-five hundred' if called directly.
+    """
+    word = ''
+    if val > 0:
+        for (dcap, dval) in (
+                (k, 100 + (100 * v)) for (v, k) in enumerate(hundreds_br)):
+            if dval + 100 > val:
+                (mod, rem) = (val % 100, val // 100)
+                if rem > 0:
+                    word += dcap
+                    if mod > 0:
+                        word += ' e '
+                if mod > 0:
+                    word += _convert_nn_br(mod)
+                return word
+
+
+def brazil_number(val):
+    if val < 100:
+        return _convert_nn_br(val)
+    if val < 1000:
+        return _convert_nnn_br(val)
+    for (didx, dval) in ((v - 1, 1000 ** v) for v in range(len(denom_br))):
+        if dval > val:
+            mod = 1000 ** didx
+            l = val // mod
+            r = val - (l * mod)
+            if mod > 99999 and l > 2:
+                ret = _convert_nnn_br(l) + ' ' + denoms_br[didx]
+            else:
+                ret = _convert_nnn_br(l) + ' ' + denom_br[didx]
+
+            if r > 0:
+                ret = ret + ', ' + brazil_number(r)
+
+            return ret
+
+
+def amount_to_text_br(number, currency):
+    number = '%.2f' % number
+    units_name = currency
+    list = str(number).split('.')
+    start_word = brazil_number(int(list[0]))
+    end_word = brazil_number(int(list[1]))
+    cents_number = int(list[1])
+    cents_name = (cents_number > 1) and 'Centavos' or 'Centavo'
+    final_result = (start_word + ' ' + units_name +
+                    ' e ' + ' ' + end_word + ' ' + cents_name)
+    return final_result
+
+
 #-------------------------------------------------------------
 # Generic functions
 #-------------------------------------------------------------
 
-_translate_funcs = {'fr' : amount_to_text_fr, 'nl' : amount_to_text_nl}
+_translate_funcs = {
+    'fr': amount_to_text_fr,
+    'nl': amount_to_text_nl,
+    'br': amount_to_text_br,
```

Comment:
@rvalyi There are differences when we reach the billions ([bilhão vs bilião](http://duvidas.dicio.com.br/bilhao-e-biliao/)), but for most daily practical uses it's not relevant.

But that's not the point.
I'm assuming that you pass the `lang` code to the function.
If this is so, and you are working with `pt_BR` Brazilian Portuguese, the key to use in  `_translate_funcs` should be that one, instead of `br`.